packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -249,6 +249,26 @@ export function removeVM(vm: VM) {
     resetComponentStateWhenRemoved(vm);
 }
 
+function assertNotSyntheticComposedWithinNative(vm: VM) {
+    const isSynthetic =
+        vm.renderMode === RenderMode.Shadow && vm.shadowMode === ShadowMode.Synthetic;
+    if (!isSynthetic) {
+        return;
+    }
+    let ancestor = vm.owner;
+    // Walk the ancestor chain to look for any native shadow components. Any native shadow component
+    // being an ancestor of a synthetic shadow component is disallowed.
+    while (!isNull(ancestor)) {
+        assert.isFalse(
+            ancestor.renderMode === RenderMode.Shadow && ancestor.shadowMode === ShadowMode.Native,
+            `<${vm.tagName}> (preferNativeShadow=false) cannot be composed inside of <${
+                getAssociatedVM(ancestor.elm).tagName
+            }> (preferNativeShadow=true)`
```

Comment:
```suggestion
            `${getComponentTag(vm)} (preferNativeShadow=false) cannot be composed inside of ${
                getComponentTag(vm)
            } (preferNativeShadow=true)`
```

I would recommend adding more details about this error by explaining that native shadow DOM components can't have synthetic shadow DOM components. 

I am wondering if stating that the `preferNativeShadow=false` when it actually `undefined` would create more confusion than anything else.  For example: https://github.com/salesforce/lwc/pull/2360/files#diff-1a8fa344696935aa14d9c83e4948e501243e500c69ae7dafc18e0bda94028897R16

Reply:
Changed the error message text